### ISSUE TITLE
towncrier 22.12.0 is breaking our docs build

### ISSUE THREAD

**Issue by ayshih** [sunpy/sunpy#6668]
The just-released towncrier 22.12.0 is breaking our docs build with the following error:
```
/home/runner/work/sunpy/sunpy/docs/whatsnew/changelog.rst:7: CRITICAL: 'Config' object is not subscriptable
```
I haven't investigated this at all; there may be a simple fix

**Comment by Cadair** [comment]
Probably needs to be fixed in sphinx-changelog

**Comment by ayshih** [comment]
It looks like the towncrier configuration was changed from an "ungodly dictionary" to a proper class (see twisted/towncrier#429), so yeah, sphinx-changelog needs to be updated.

**Comment by dstansby** [comment]
See https://github.com/OpenAstronomy/sphinx-changelog/issues/16 for upstream issue

**Comment by ayshih** [comment]
The sphinx-changelog 1.3.0 release has this fixed, and towncrier was unpinned in #6813
